File: Práticas/C++/#01/daa003.cpp
#include <iostream>
#include <cstdlib>
#include <cstring>

using namespace std;

int id_geracao(char s[]){
  int cases = 5290000;

  if(isalpha(s[0])){
    if(isalpha(s[7]))
      return cases * 3;

    else
      return 0;
  }

  else{
    if(isalpha(s[3]))
      return cases * 2;

    else
      return cases;
  }
}

int number(char s[]){
  int num = id_geracao(s);
  int base = 1;

  for(int i=7; i>=0; i--){
    if(isdigit(s[i])){
      num += (s[i] - '0') * base;
      base *= 10;
    }
  }

  for(int i=7; i>=0; i--){
    if(isalpha(s[i])){
      int value = s[i] - 'A';

      if(s[i] > 'K')
	value--;

      if(s[i] > 'W')
	value--;

      if(s[i] > 'Y')
	value--;

      num += value * base;
      base *= 23;
    }
  }

  return num;
}

int main(){
  int n;

  cin >> n;

  char str[20];

  fgets(str, 20, stdin);
  
  for(int i=0; i<n; i++){
    fgets(str, 20, stdin);
    
    char *str1;
    char *str2;

    str1 = strtok(str, " ");
    str2 = strtok(NULL, "\n");

    cout << abs(number(str1) - number(str2)) << endl;
  }

  return 0;
}

File: Lab Classes/Java/#01/daa003.java
import java.util.Scanner;
import java.lang.Math;

public class daa003{
    static int id_geracao(String s){	
        int cases = 5290000;

	if(Character.isLetter(s.charAt(0))){
	    if(Character.isLetter(s.charAt(7)))
		return cases*3;

	    else
		return 0;
	}

	else{
	    if(Character.isLetter(s.charAt(3)))
		return cases*2;

	    else
		return cases;
	}
    }

    static int number(String s){
	int num = id_geracao(s);
	int base = 1;

	for(int i=7; i>=0; i--){
	    if(Character.isDigit(s.charAt(i))){
		num += (s.charAt(i) - '0') * base;
		base *= 10;
	    }
	}

	for(int i=7; i>=0; i--){
	    if(Character.isLetter(s.charAt(i))){
		int value = s.charAt(i) - 'A';

		if(s.charAt(i) > 'K')
		    value--;

		if(s.charAt(i) > 'W')
		    value--;

		if(s.charAt(i) > 'Y')
		    value--;

		num += value * base;
		base *= 23;
	    }
	}

	return num;
    }
    
    static int diferenca(String s, String s1){
	int diff = Math.abs(number(s) - number(s1));
	
	return diff;
    }
   

    public static void main(String[] args){
	Scanner in = new Scanner(System.in);
	int n = in.nextInt();
	

	for(int i=0; i<n; i++){
	    String str1 = in.next();
	    String str2 = in.next();

	    //System.out.println("Número da primeira matrícula: " + number(str1));
	    //System.out.println("Número da segunda matrícula: " + number(str2));
	    System.out.println(diferenca(str1, str2));
	    //System.out.println();
	}    
    }
}
